routes/degree_days.py

```diff
@@ -0,0 +1,375 @@
+    """
+
+    point_pkg = {}
+    if horp == "all":
+        for mi, v_li in enumerate(point_data):  # (nested list with model at dim 0)
+            if mi == 0:
+                min_year = years_lu["historical"]["min"]
+                max_year = years_lu["historical"]["max"]
+                years = range(min_year, max_year + 1)
+            else:
+                min_year = years_lu["projected"]["min"]
+                max_year = years_lu["projected"]["max"]
+                years = range(min_year, max_year + 1)
+
+            model = dim_encodings["model"][mi]
+            point_pkg[model] = {}
+
+            # Responses from Rasdaman include the same array length for both
+            # historical and projected data, representing every possible year
+            # (1979-2100). This means both the historical and projected data
+            # arrays include nodata years populated with 0s. The code below
+            # omits nodata gaps and makes sure the correct year is assigned to
+            # its corresponding data in the historical and projected data
+            # arrays.
+            year = years_lu["historical"]["min"]
+            year_index = 0
+            for value in v_li:
+                if year in years:
+                    point_pkg[model][years[year_index]] = {"dd": value}
+                    year_index += 1
+                year += 1
+    else:
+        if horp in ["historical", "hp"]:
+            historical_max = round(point_data[0], 1)
+            historical_mean = round(point_data[1], 1)
+            historical_min = round(point_data[2], 1)
+
+            point_pkg["historical"] = {}
+            point_pkg["historical"]["ddmin"] = historical_min
+            point_pkg["historical"]["ddmean"] = historical_mean
+            point_pkg["historical"]["ddmax"] = historical_max
+
+        if horp in ["projected", "hp"]:
+            if horp == "projected":
+                projected_max = round(point_data[0], 1)
+                projected_mean = round(point_data[1], 1)
+                projected_min = round(point_data[2], 1)
+            else:
+                projected_max = round(point_data[3], 1)
+                projected_mean = round(point_data[4], 1)
+                projected_min = round(point_data[5], 1)
+
+            point_pkg["projected"] = {}
+            point_pkg["projected"]["ddmin"] = projected_min
+            point_pkg["projected"]["ddmean"] = projected_mean
+            point_pkg["projected"]["ddmax"] = projected_max
+
+    return point_pkg
+
+
+async def fetch_dd_point_data(x, y, cov_id, horp, start_year, end_year):
+    """Run the async degree days data request for a single point
+    and return data as json
+
+    Args:
+        lat (float): latitude
+        lon (float): longitude
+        cov_id (list): string of heating_degree_days, degree_days_below_zero,
+            thawing_index, or freezing_index
+        horp [Historical or Projected] (str): historical, projected, or all
```

Comment:
minor nitpick, include "hp" option in horp arg line of dosctring

Reply:
Fixed in 352c2027d213a687a5433fad95548ea0e6ba52fd.